bundles/GraphicServiceBilling/Service/GraphicServiceBillingService.php

```diff
@@ -54,30 +54,141 @@ public function __construct(
      * @param $tasks
      *
      * @return array
+     *
+     * @throws \Billing\Exception\InvoiceException
      */
-    public function createExportData($tasks)
+    public function createExportDataMarketing($tasks)
     {
         // Get debtor custom field.
-        $debtorCustomFieldId = $this->billingService->getCustomFieldId('Debitor');
+        $debtorFieldId = $this->billingService->getCustomFieldId('Debitor');
+        $marketingAccountFieldId = $this->billingService->getCustomFieldId('Marketing Account');
+        $libraryFieldId = $this->billingService->getCustomFieldId('Library');
 
         $entries = [];
 
         foreach ($tasks as $task) {
-            // Strip file link and \\ from description.
-            $description = $task->fields->description;
-            $description = preg_replace('/\\\\ \[Åbn filer i OwnCloud.*]\\ /i', '', $description);
-            $description = preg_replace('/\\\\/', '', $description);
+            $marketingAccount = $task->fields->{$marketingAccountFieldId} ?? false;
+
+            if ('Markedsføringskonto' !== $marketingAccount[0]->value) {
+                continue;
+            }
+
+            $description = '';
+
+            $library = $task->fields->{$libraryFieldId};
+
+            if (isset($entries[$library])) {
+                $header = $entries[$library]->header;
+            } else {
+                // Create header line data.
+                $header = (object) [
+                    'debtor' => $this->boundReceiverAccount,
+                    'salesChannel' => '10',
```

Comment:
Should this (`salesChannel`) be configurable?

Reply:
Not in this case.